### PR TITLE
Provide tracing-datadog-addr option

### PR DESCRIPTION
**Pull request by GaryPWhite**
Referencing recent agent release (3.27.0) and https://github.com/buildkite/agent/pull/1273

### CODE REVIEW

**Comment by plaindocs** [comment]
Hey @GaryPWhite thanks for adding this. 

What do you think about adding a brief description instead of linking the GitHub issue? And maybe adding the environment variable and backlinking from https://buildkite.com/docs/pipelines/environment-variables ?

**Comment by GaryPWhite** [comment]
I definitely will -- I think I may need to edit this branch as now I'm seeing the original post on the PR may not match the code changes -- stand by 😄

**Comment by GaryPWhite** [comment]
yep -- looks like the option was changed to either `BUILDKITE_TRACING_BACKEND` or `tracing-backend` -- by default it would be `""` but can be set to `datadog` in order to send metrics to `127.0.0.1:8126` -- i'll update the docs.

**Comment by GaryPWhite** [comment]
will add a more meaningful description / env var when I confirm this configuration works!

**Comment by GaryPWhite** [comment]
alrighty @plaindocs thanks for the input! put in a more useful decription and environment variable. Can confirm they work 👍

**Review by chloeruka — APPROVED** [review]
Nice! Thank you for adding this 😊

File: pages/agent/v3/configuration.md.erb [hunk 1]
```diff
@@ -417,6 +417,17 @@ _Optional attributes:_
...
+      <th><code>tracing-backend</code></th>
+      <td>
+        Indicates if tracing should be supplied for builds/jobs with Datadog APM, or not at all. When set to Datadog backend, will use default APM address `localhost:8126`, or provided (Datadog) variables like `DD_AGENT_HOST` and `DD_AGENT_APM_PORT`.        
+        Accepts blank string or "datadog".
+        <p class="Docs__api-param-eg"><em>Default:</em> <code>""</code></p>
+        <p class="Docs__api-param-eg"><em>Environment variable:</em> <code>BUILDKITE_TRACING_BACKEND</code></p>
+      </td>
+    </tr>
+
+
```
Comment (plaindocs) [inline pages/agent/v3/configuration.md.erb:430]:
```suggestion
    <tr>
      <th><code>tracing-backend</code></th>
      <td>
        Set to <code>datadog</code> to enable Datadog API tracing for builds using the default APM address <code>localhost:8126</code>, or Datadog variables like <code>DD_AGENT_HOST</code> and <code>DD_AGENT_APM_PORT<code>.        
        Accepts blank string or "datadog".
        <p class="Docs__api-param-eg"><em>Default:</em> <code>""</code></p>
        <p class="Docs__api-param-eg"><em>Environment variable:</em> <code>BUILDKITE_TRACING_BACKEND</code></p>
      </td>
    </tr>

    
```

File: pages/pipelines/environment_variables.md.erb [hunk 2]
```diff
@@ -517,6 +517,14 @@ A colon separated list of non-private team names that the user who unblocked the
 
 Example: `"everyone:platform"`
 
+<h3 class="h3-caps">BUILDKITE_TRACING_BACKEND</h3>
+
+Set to either `""` or `"datadog"` to send metrics nowhere, or to [Datadog APM](https://docs.datadoghq.com/tracing/). When set to `"datadog"`, will send metrics to `localhost:8126`, or `DD_AGENT_HOST` and `DD_AGENT_APM_PORT`.
+
+Defaults to empty string. Also available as a [buildkite agent configuration option.](/docs/agent/v3/configuration#configuration-settings)
```
Comment (plaindocs) [inline pages/pipelines/environment_variables.md.erb:524]:
```suggestion
Set to `"datadog"` to send metrics to the [Datadog APM](https://docs.datadoghq.com/tracing/) via `localhost:8126`, or `DD_AGENT_HOST:DD_AGENT_APM_PORT`.

Defaults to empty string. Also available as a [buildkite agent configuration option.](/docs/agent/v3/configuration#configuration-settings)
```

**Review by plaindocs — COMMENTED** [review]
Thanks again for helping get this out. I've made a couple of suggested changes which I think are more clear.

**Comment by GaryPWhite** [comment]
committed changes! Thanks for the help, folks!

**Pull request merged**